src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/InitiateTaskDbLockAndTransactionTest.java

```diff
@@ -0,0 +1,282 @@
+
+    @BeforeEach
+    void setUp() {
+        taskId = UUID.randomUUID().toString();
+
+        taskManagementService = new TaskManagementService(
+            camundaService,
+            camundaQueryBuilder,
+            permissionEvaluatorService,
+            cftTaskDatabaseService,
+            cftTaskMapper,
+            launchDarklyFeatureFlagProvider,
+            configureTaskService,
+            taskAutoAssignmentService
+        );
+
+        lenient().when(launchDarklyFeatureFlagProvider.getBooleanValue(
+                           RELEASE_2_CANCELLATION_COMPLETION_FEATURE,
+                           IDAM_USER_ID
+                       )
+        ).thenReturn(true);
+
+        testTaskResource = new TaskResource(taskId, A_TASK_NAME, A_TASK_TYPE, ASSIGNED);
+        when(taskAutoAssignmentService.autoAssignCFTTask(any(TaskResource.class)))
+            .thenReturn(testTaskResource);
+
+        when(configureTaskService.configureCFTTask(any(TaskResource.class), any(TaskToConfigure.class)))
+            .thenReturn(testTaskResource);
+    }
+
+    @AfterEach
+    void tearDown() {
+        taskResourceRepository.deleteAll();
+    }
+
+    @Test
+    void given_initiate_task_is_called_when_error_then_rollback_db() {
+        when(taskAutoAssignmentService.autoAssignCFTTask(any(TaskResource.class)))
+            .thenThrow(new RuntimeException("some error"));
+
+        assertThrows(RuntimeException.class, () -> transactionHelper
+            .doInNewTransaction(() -> taskManagementService.initiateTask(taskId, initiateTaskRequest)));
+
+        transactionHelper.doInNewTransaction(() -> assertEquals(0, taskResourceRepository.count()));
+
+        verify(cftTaskMapper).mapToTaskResource(taskId, initiateTaskRequest.getTaskAttributes());
+        verify(cftTaskDatabaseService).saveTask(argThat((task) -> task.getTaskId().equals(taskId)));
+        verifyNoMoreInteractions(cftTaskDatabaseService);
+    }
+
+    @Test
+    void given_task_is_not_locked_when_initiated_task_is_called_then_it_succeeds() {
+        transactionHelper.doInNewTransaction(() -> taskManagementService.initiateTask(taskId, initiateTaskRequest));
+
+        InOrder inOrder = inOrder(
+            cftTaskMapper,
+            cftTaskDatabaseService,
+            configureTaskService,
+            taskAutoAssignmentService,
+            camundaService,
+            cftTaskDatabaseService
+        );
+
+        inOrder.verify(cftTaskMapper).mapToTaskResource(taskId, initiateTaskRequest.getTaskAttributes());
+
+        inOrder.verify(cftTaskDatabaseService).saveTask(taskResourceCaptor.capture());
+        assertTrue(expectTaskWithAssignee(taskResourceCaptor.getValue()));
+
+        inOrder.verify(configureTaskService).configureCFTTask(
+            taskResourceCaptor.capture(),
+            eq(new TaskToConfigure(taskId, A_TASK_TYPE, null, A_TASK_NAME))
+        );
+        assertTrue(expectTaskWithAssignee(taskResourceCaptor.getValue()));
+
```

Comment:
Can we also assert the CFTTaskState to show it is configured? i.e. check the state is unassigned as per AC 3